pkg/ocm/clusters.go

```diff
@@ -604,3 +639,44 @@ func (c *Client) ResumeCluster(clusterID string) error {
 
 	return nil
 }
+
+//Get all the archived ROSA STS clusters
+func (c *Client) GetArchivedCluster(clusterKey string) (clusters []*cmv1.Cluster, err error) {
+	query := fmt.Sprintf("id = '%s' OR name = '%s' OR external_id = '%s'",
+		clusterKey, clusterKey, clusterKey,
+	)
+	request := c.ocm.ClustersMgmt().V1().ArchivedClusters().List().Search(query)
+	response, err := request.Send()
+
+	if err != nil {
+		if response.Status() == http.StatusNotFound {
+			return clusters, handleErr(response.Error(), err)
+		}
+		return clusters, err
+	}
+	response.Items().Each(func(cluster *cmv1.Cluster) bool {
+		clusters = append(clusters, cluster)
+		return true
+	})
+
+	return clusters, nil
+}
+
+func (c *Client) ListAccountArchivedClusters(accountID string) ([]*cmv1.Cluster, error) {
+	var clusters []*cmv1.Cluster
+
+	query := fmt.Sprintf("product.id like '%%rosa%%' and aws.sts.role_arn like '%%%s%%'", accountID)
+	request := c.ocm.ClustersMgmt().V1().ArchivedClusters().List().Search(query)
+
+	response, err := request.Send()
+	if err != nil {
+		return clusters, err
+	}
+
+	response.Items().Each(func(cluster *cmv1.Cluster) bool {
+		clusters = append(clusters, cluster)
+		return true
+	})
+
+	return clusters, nil
+}
```

Comment:
This function does not fetch all clusters, only the first 100 in the database or so. Can we change this? I tested it and it failed to find a cluster that I created and deleted last night. (It worked if I specificied the `-c $id`)